### PR TITLE
Williamhe/1410386 add pipeline curl retries

### PR DESCRIPTION
**Pull request by william-msft**
- [ ] The purpose of this PR is explained in this message or in an issue. If an issue please include a reference as \#\<issue\_number\>.
- [ ] Tests are included and/or updated for code changes.
- [ ] Proper license headers are included in each file.

### CODE REVIEW

File: images/__common.sh [hunk 1]
```diff
@@ -1,5 +1,6 @@
 __CURRENT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
 source $__CURRENT_DIR/__sdkStorageConstants.sh
+source $__CURRENT_DIR/build/__common.sh
```
Comment (arroyc) [inline images/__common.sh:3]: we already have a common.sh, do not add another common.sh. Add your function to the existing common.sh

File: images/build/Dockerfiles/gitHubActions.Dockerfile [hunk 2]
```diff
@@ -62,13 +62,16 @@ COPY --from=buildscriptgenerator /opt/buildscriptgen/ /opt/buildscriptgen/
 ARG BUILD_DIR="/opt/tmp/build"
 ARG IMAGES_DIR="/opt/tmp/images"
 RUN ${IMAGES_DIR}/build/installHugo.sh
+RUN "source ${BUILD_DIR}/build/__common.sh"
```
Comment (arroyc) [inline images/build/Dockerfiles/gitHubActions.Dockerfile:65]: have another script (say foo.sh) which will use the function from common.sh. inside foo.sh you need to source common.sh.

nit: do not add new RUN steps as it increases layers, source it between line 73 and 74

File: images/build/Dockerfiles/gitHubActions.Dockerfile [hunk 3]
```diff
@@ -62,13 +62,16 @@ COPY --from=buildscriptgenerator /opt/buildscriptgen/ /opt/buildscriptgen/
...
 RUN ${IMAGES_DIR}/build/installHugo.sh
+RUN "source ${BUILD_DIR}/build/__common.sh" 
+# COPY requests.sh somwhere
 RUN set -ex \
  && yarnCacheFolder="/usr/local/share/yarn-cache" \
  && mkdir -p $yarnCacheFolder \
  && chmod 777 $yarnCacheFolder \
+ # TODO: invoke requests.sh
  && . ${BUILD_DIR}/__nodeVersions.sh \
  && ${IMAGES_DIR}/receiveGpgKeys.sh 6A010C5166006599AA17F08146C2130DFD2497F5 \
```
Comment (arroyc) [inline images/build/Dockerfiles/gitHubActions.Dockerfile:73]: nit: we do have a retry logic in receivegpgkeys.sh, you can refer to it as well

**Review by arroyc — CHANGES_REQUESTED** [review]
make sure build works and tests are succesful

File: images/build/Dockerfiles/gitHubActions.Dockerfile [hunk 4]
```diff
@@ -60,15 +60,15 @@ FROM main AS intermediate
...
 RUN ${IMAGES_DIR}/build/installHugo.sh
 RUN set -ex \
  && yarnCacheFolder="/usr/local/share/yarn-cache" \
  && mkdir -p $yarnCacheFolder \
  && chmod 777 $yarnCacheFolder \
  && . ${BUILD_DIR}/__nodeVersions.sh \
  && ${IMAGES_DIR}/receiveGpgKeys.sh 6A010C5166006599AA17F08146C2130DFD2497F5 \
- && curl -fsSLO --compressed "https://yarnpkg.com/downloads/$YARN_VERSION/yarn-v$YARN_VERSION.tar.gz" \
+ && ${IMAGES_DIR}/retry.sh "curl -fsSLO --compressed https://yarnpkg.com/downloads/$YARN_VERSION/yarn-v$YARN_VERSION.tar.gz" \
  && curl -fsSLO --compressed "https://yarnpkg.com/downloads/$YARN_VERSION/yarn-v$YARN_VERSION.tar.gz.asc" \
```
[the remark this answers is not in the document]
Comment (william-msft) [reply images/build/Dockerfiles/gitHubActions.Dockerfile:72]: Yes, there'll be even more places to add too. Currently the draft PR is to test the one above while I debug:
https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=5290181&view=logs&j=3b2a2d55-d8d9-5a78-6715-6219efc4cdf5&t=ea2fb0ea-8c3b-5cd8-4c59-17ddeb174606&l=46

**Review by arroyc — APPROVED** [review]
make sure you run a succesful nightly build to ensure dynamic build scripts are fine

**Pull request merged**